packages/auth-nextjs/src/shared.ts

```diff
@@ -153,8 +159,9 @@ export abstract class NextAuth extends NextAuthHelpers {
     return {
       GET: async (
         req: NextRequest,
-        { params }: { params: { auth: string[] } },
+        ctx: { params: Promise<{ auth: string[] }> },
```

Comment:
Will this work in <15 or will those users need to do a type-cast?

Reply:
Yeah, I'm not worried too much about runtime, and even if it breaks backwards compatibility, I'm fine with shipping it, I just want to understand everything that might break here so we can update the docs.

Reply:
Well the `GET` is exported in a `route.ts` to act as a Next.js Route Handler but it's not type checked against anything. I can run and build the project running Next.js before their breaking change with this version, no errors. And i don't know of any way that someone could be type-checking this and get a warning/error after this change.

At most we could add a small note to the docs referencing the change introduced by Next.js 15 and how this small library targets the new behaviour but supports both.